File: backend/src/main/java/com/projects/vidyara/backend/library/dto/DocumentResponse.java
package com.projects.vidyara.backend.library.dto;

import com.projects.vidyara.backend.library.enums.DocumentStatus;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Getter
@Setter
public class DocumentResponse {
    private UUID id;

    private String title;

    private String description;

    private String fileType;

    private Long fileSize;

    private UUID uploadedBy;

    private DocumentStatus status;

    private Instant createdAt;
}

File: frontend/src/models/DocumentResponse.ts
export type DocumentStatus = 'PENDING' | 'PROCESSING' | 'APPROVED' | 'REJECTED';

export default interface DocumentResponse {
  id: string;
  title: string;
  description?: string;
  fileType?: string;
  fileSize?: number;
  uploadedBy?: string;
  status: DocumentStatus;
  createdAt?: string;
}
